feat: first iteration of task

suivo2001 opened the pull request.

Commit:
feat: first iteration of task

Commit:
fix: delete test image

mcmcgrath13 commented on package.json:
```diff
@@ -1,17 +1,17 @@
 {
-  "name": "honeycomb",
+  "name": "task-ratings",
   "description": "all-in-one task starter app with jsPsych + React + Electron + psiturk ",
```
update the description

mcmcgrath13 commented on src/App.jsx:
```diff
@@ -41,7 +42,8 @@ function App() {
   };
   const desktopUpdateFunction = (data) => {
-    ipcRenderer.send("data", data);
+    if (data.trial_type!=='image_keyboard_response')
```
why not send the image keyboard response trial data?

mcmcgrath13 commented on src/timelines/main.js:
```diff
@@ -1,33 +1,28 @@
 ];
 
+for (let i = 5; i >= 0; i--) {
+  primaryTimeline.push(countdown(300*(i+1), 300*i));
```
why does the countdown vary with the trial number? shouldn't this be the same block repeated 5 times?

suivo2001 replied:
Yunshu said she wanted the countdown from total time for each block, so 30 minutes to 25, then 25 to 20, ...

mcmcgrath13 commented on src/trials/countdown.js:
```diff
@@ -0,0 +1,21 @@
+    
+    return {
+        type: 'image_keyboard_response',
+        stimulus: blank,
```
why use a blank image instead of an empty div or similar?

mcmcgrath13 commented on src/trials/enterRating.js:
```diff
@@ -0,0 +1,17 @@
+        questions: [{ prompt: baseStimulus(`<h1>${lang.instructions.enter_rating}</h1>`, true)}],
+        on_finish: (data) => {
+            data.submit_time = Date.now()
+            jsPsych.data.write(data)
```
shouldn't need the `jsPsych.data.write`, just updating data here should do the trick.

Commit:
feat: new countdown method

mcmcgrath13 commented on src/App.jsx:
```diff
@@ -42,7 +42,7 @@ function App() {
   };
   const desktopUpdateFunction = (data) => {
-    if (data.trial_type!=='image_keyboard_response')
+    if (data.countdown!==true)
```
delete the blank image from assets and add a comment on the timeline about the countdown math, otherwise looks good

Commit:
style: remove unused image and comment on countdown

Commit:
fix: remove images from config

Commit:
fix: remove unused stuff

mcmcgrath13 commented on package.json:
```diff
@@ -1,17 +1,17 @@
-  "name": "honeycomb",
-  "description": "all-in-one task starter app with jsPsych + React + Electron + psiturk ",
+  "name": "task-ratings",
+  "description": "ratings task for OCD Patient study",
```
does this repo need to be private? (cc @mirestrepo )

mirestrepo replied:
Nope

mcmcgrath13 commented on src/lib/taskUtils.js:
```diff
@@ -12,7 +12,23 @@ const generateStartingOpts = (blockSettings) => {
+	o.stop(context.currentTime + 1)
+}
+
+const sleep = (ms) => {
```
sleep is already defined in `utils` - import the function from there

Commit:
refactor: use sleep from util

mcmcgrath13 reviewed (approved):
I'd have @mirestrepo approve this one too as she has a better sense of the task requirements

mirestrepo commented on src/trials/enterRating.js:
```diff
@@ -0,0 +1,15 @@
+const enterRating = () => {
+
+    return {
+        type: 'survey_text',
```
I feel we should checking with Yunshu on this. Seems that having a free text field for 1-10 is error prone. I wonder if [jspsych-html-slider-response](https://www.jspsych.org/plugins/jspsych-html-slider-response/) is better

mirestrepo reviewed (approved):
I think it's okay to merge this as a first version. There are few questions for Yunshu.

1. Type of field for the 1-10 rating
2. Should we add some padding to the instructions? They seem pretty close to the edged, are other tasks like that?
3. Should the timer be centered or on the top as it is

mirestrepo reviewed (changes requested):
Change the 1-10 input field to multiple select buttons. I think https://www.jspsych.org/plugins/jspsych-survey-multi-choice/ is the one...

Commit:
feat: change rating to multiple choice

suivo2001 commented:
Changed rating to multiple choice, centered the timer.

mirestrepo commented on src/App.css:
```diff
@@ -46,7 +46,7 @@ html, body, #root, .App {
 
 .main-prompt, .center_container {
-  height: 100vh;
+  height: 95vh;
```
Can we also adjust the width?

mirestrepo reviewed (approved):
The css is funky... I'm not sure if it's easy to adjust but the radio buttons are super close to the numbers, and one needs to scroll to see the continue button, if any of those can be changed that would be great. But if not, you can merge and we can ask Yunshu more

mirestrepo commented on src/language/en_us.json:
```diff
@@ -26,96 +26,14 @@
-    }
+    "check_event_marker": "Make sure the event marker is connected and then continue.",
+    "main_instructions": "For the next 30 min, we will record your brain activity while you are going about your daily activities. Please keep the tablet in the same room as you, and wear the CTM and the sling. Every 5 min, there will be a beep sound and you will be asked to submit your OCD symptom intensity using your keyboard. Please adjust the volume to 50/100. When you are ready, press the bu ...
+    "enter_rating": "Please rate your OCD symptom intensity from 0-10. Press continue to submit your rating.”"
```
There is an extra closing " that is not needed here

Commit:
fix: adjust css

Commit:
style: fix text

suivo2001 commented:
I added a margin for the radio buttons, adjusted the full width and the instructions text. I'm pretty sure when the app is in fullscreen, which the participants should be running the task at, there's plenty of space between the continue button and the bottom, so I don't think its an issue.

mirestrepo reviewed (approved):
Looks great!
